plugins/unocss.ts

```diff
@@ -1,72 +1,133 @@
-import unocss, { presetUno } from "../deps/unocss.ts";
-import { getExtension } from "../core/utils/path.ts";
 import { merge } from "../core/utils/object.ts";
+import { read } from "../core/utils/read.ts";
+import { createGenerator, presetUno, resetUrl } from "../deps/unocss.ts";
 
-import type { UserConfig } from "../deps/unocss.ts";
 import type Site from "../core/site.ts";
+import type {
+  PluginOptions,
+  UnocssPluginContext,
+  UserConfig,
+} from "../deps/unocss.ts";
 
 export interface Options {
-  /** Extensions processed by this plugin to extract the utility classes */
-  extensions?: string[];
-
   /**
-   * Options passed to UnoCSS.
-   * @see https://unocss.dev/guide/config-file
+   * Configurations for UnoCSS.
+   * @see {@link https://unocss.dev/guide/config-file}
+   */
+  config?: UserConfig;
+  /**
+   * Set the css filename for all generated styles,
+   * Set to `false` to insert a <style> tag per page.
+   * @defaultValue `false`
+   */
+  cssFile?: false | string;
+  /**
+   * Process CSS files using UnoCSS transformers.
+   * @defaultValue `undefined`
+   */
+  cssFileTransformers?: PluginOptions["transformers"];
+  /**
+   * Supported CSS reset options.
+   * @see {@link https://github.com/unocss/unocss/tree/main/packages/reset}
+   * @defaultValue `tailwind`
    */
-  options?: Omit<UserConfig, "content">;
+  reset?: false | "tailwind" | "tailwind-compat" | "eric-meyer";
 }
 
 export const defaults: Options = {
-  extensions: [".html"],
-  options: {
+  config: {
     presets: [presetUno()],
   },
+  cssFile: false,
+  reset: "tailwind",
 };
 
 export default function (userOptions?: Options) {
   const options = merge(defaults, userOptions);
 
   return (site: Site) => {
-    // deno-lint-ignore no-explicit-any
-    let unoPlugins: any[];
+    const uno = createGenerator(options.config);
 
-    if (site.hooks.postcss) {
-      throw new Error(
-        "PostCSS plugin is required to be installed AFTER UnoCSS plugin",
-      );
+    if (Array.isArray(options.cssFileTransformers)) {
+      site.process([".css"], async (files) => {
+        const { default: MagicString } = await import(
+          "npm:magic-string@0.30.5"
```

Comment:
Please, move this import to `deps/unocss.ts`.

Reply:
Since this is an optional dependency, my thinking is that users don't have to download it if they don't need this feature.